hibernate-core/src/test/java/org/hibernate/orm/test/annotations/manytoone/ManyToOneJoinTest.java

```diff
@@ -36,18 +37,9 @@ public class ManyToOneJoinTest {
 	public void teardDown(SessionFactoryScope scope) {
 		scope.inTransaction(
 				session -> {
-					List<ForestType> forestTypes = session.createQuery( "from ForestType" ).list();
-					forestTypes.forEach(
-							forestType -> {
-								forestType.getTrees().forEach(
-										tree ->{
-											session.delete( tree.getForestType() );
-											session.delete( tree );
-										}
-								);
-								session.delete( forestType );
-							}
-					);
+					session.createMutationQuery( "delete from TreeType" ).executeUpdate();
+					session.createMutationQuery( "delete from ForestType" ).executeUpdate();
+					session.createMutationQuery( "delete from BiggestForest" ).executeUpdate();
 				}
 		);
```

Comment:
Can replace all this with:

```java
		scope.getSessionFactory().getSchemaManager().truncateMappedObjects();
```

Reply:
The session factory already truncates everything when the test finishes, but usually we explicitly clear the test data before.